index.js

```diff
@@ -1,48 +1,20 @@
-var findInBatches = require('find-in-batches')
 var request = require('request')
-var cheerio = require('cheerio')
 
-exports.search = function (query, options, callback) {
-  if (arguments.length === 2) {
-    callback = options
-    options = {}
-  }
+exports.search = function(query, options, callback) {
+    if (arguments.length === 2) {
+        callback = options
+        options = {
+            limit: 20
+        }
+    }
 
-  query = (query || '').trim()
-  if (!query) return callback(null, [])
-
-  var opts = { batchSize: 20, maximum: 20 }
-  if (options.limit) opts.maximum = options.limit
-
-  var modules = []
-  findInBatches.each(opts, function find (options, callback) {
-    search(query, options, callback)
-  }, function each (m, done) {
-    modules.push(m)
-    done()
-  }, function (err) {
-    if (err) return callback(err)
-    callback(null, modules)
-  })
-}
-
-function search (query, options, callback) {
-  var page = options.page || 1
-  return request({
-    method: 'get',
-    url: 'https://www.npmjs.com/search?page=' + page + '&q=' + query
-  }, function (err, res, body) {
-    if (err) return callback(err)
-    var $ = cheerio.load(body)
-    var modules = []
-    $('.package-details').each(function () {
-      modules.push({
-        name: $(this).find('.packageName').text(),
-        version: $(this).find('.version').text(),
-        author: $(this).find('.authorName').text(),
-        description: $(this).find('.description').text()
-      })
+    request({
+        method: 'get',
+        url: 'https://www.npmjs.com/search/suggestions?q=' + query + '&size=' + options.limit,
+        headers: { 'Content-Type': 'application/json' }
```

Comment:
I'd prefer `json: true` as option as that also parses the response and catches `JSON.parse` errors.
You could remove the `JSON.parse` call on line 18 as soon as you pass the option.